### PR TITLE
Migrate Docker registry from Docker Hub to GitHub container registry

### LINKED ISSUE #823: Migrate Central Dogma container registry
Central Dogma publishes the server image to Docker Hub.
https://github.com/line/centraldogma/blob/087cb85fa79702220dc56b8cb9fd78d921bad76c/.github/workflows/publish-docker.yml#L42-L51
Unfortunately, Docker is going to stop support for open source organisations.
https://blog.alexellis.io/docker-is-deleting-open-source-images/

GitHub container registry could be an alternative. 
https://github.blog/2020-09-01-introducing-github-container-registry/

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

Although Docker decided to [revert sunset the “Docker Free Team” plan](https://www.docker.com/blog/we-apologize-we-did-a-terrible-job-announcing-the-end-of-docker-free-teams/),
we still think it is worth migrating our container registry to GitHub.
Because users will easily check out the images on our GitHub repository.

Modifications:

- Fixed to publish docker images to `ghcr.io`

Result:

- You can now get Central Dogma docker images from `ghrc.io`
```sh
docker pull ghrc.io/line/centraldogma:latest
```
- Closes #823

### CODE REVIEW

**Comment by ikhoon** [comment]
https://github.com/ikhoon/centraldogma/pkgs/container/centraldogma This is an example registry that was created by this change.

**Comment by ikhoon** [comment]
@KarboniteKream I think this change will affect your test environment.

File: dist/build.gradle [hunk 1]
```diff
@@ -153,7 +153,7 @@ task docker(group: 'Build',
             dependsOn: tasks.distDir,
             type: DockerBuildImage) {
     inputDir = file('.')
-    def name = 'line/centraldogma:'
+    def name = 'ghcr.io/line/centraldogma:'
```
Comment (jrhee17) [inline dist/build.gradle:156]: Question) I understood this task to just build the image locally, so I find it a little odd that the registry name is included in the image name. Was this change necessary?

**Review by jrhee17 — APPROVED** [review]
Left a minor question but looks good! Thanks @ikhoon 🙇 👍 🙇

File: dist/build.gradle (hunk 1 above)
Comment (ikhoon) [reply dist/build.gradle:156]: Without the registry prefix, the following command fails to find the target image.
```
docker push ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest 
```
So, it is necessary.

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by trustin — APPROVED** [review]
Thanks a lot for taking care of this, @ikhoon ! 🙇
